### PR TITLE
Fixes #9725: Data source Scala model

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/9725

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+sealed trait DataSourceType {
+  def name : String
+}
+
+case object ByNodeSourceType extends DataSourceType {
+  val name = "byNode"
+}
+
+case object GlobalSourceType extends DataSourceType {
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala]: This should be renamed to "AllNodeSourceType"

File: rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  val name = "global"
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala]: Here, we should use a case class with two more parameters: 

- identifyingAttributePath: the Json Path that identifies the attribute used to know what node the json is for; 
- rudderNodeProperty: the "${node.property[foo][bar]}" or "${rudder.node.hostname}" to compare identifyingAttributePath against to.

File: rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+case class DataSourceName(value : String) extends AnyVal
+
+case class DataSource (
+    name       : DataSourceName
+  , description: String
+  , sourceType : DataSourceType
+  , url        : String
+  , headers    : Map[String,String]
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala]: perhaps we should add a "httpMethod", even if we only authorize GET for now.

File: rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  , path       : String
+  , frequency  : Int
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala]: Do you think changing type to "joda.time.Seconds" would be interesting ?

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**